iroh-net/src/endpoint.rs

```diff
@@ -142,6 +149,26 @@ impl Builder {
 
     // # The very common methods everyone basically needs.
 
+    /// Set the IPv4 bind address.
+    ///
+    /// Setting the port to `0` will use a random port.
+    ///
+    /// By default will use `0.0.0.0:0` to bind to.
```

Comment:
Could we add a line describing that we fall back to port 0 if the specified port is already in use?  Because this is kind of surprisingly different from classical behaviour.  Same for IPv6.

Reply:
updated docs some more